initializr-generator/src/main/groovy/io/spring/initializr/generator/ProjectRequest.groovy

```diff
@@ -78,14 +78,30 @@ class ProjectRequest extends BasicProjectRequest {
 	 */
 	void resolve(InitializrMetadata metadata) {
 		List<String> depIds = style ? style : dependencies
+		List<String> additionalDepIds = new ArrayList<>()
 		String actualBootVersion = bootVersion ?: metadata.bootVersions.default.id
 		Version requestedVersion = Version.parse(actualBootVersion)
 		resolvedDependencies = depIds.collect {
 			def dependency = metadata.dependencies.get(it)
 			if (dependency == null) {
 				throw new InvalidProjectRequestException("Unknown dependency '$it' check project metadata")
 			}
-			dependency.resolve(requestedVersion)
+			dependency = dependency.resolve(requestedVersion)
```

Comment:
I agree, when I first wrote this. I had the dependencies hidden, but the architect's wanted full transparency to the user base. Let me see if I can rework this to fix both our purposes.